packages/next/build/webpack/plugins/next-trace-entrypoints-plugin.ts

```diff
@@ -0,0 +1,238 @@
+import nodePath from 'path'
+import { nodeFileTrace } from 'next/dist/compiled/@vercel/nft'
+import {
+  webpack,
+  isWebpack5,
+  sources,
+} from 'next/dist/compiled/webpack/webpack'
+import { TRACE_OUTPUT_VERSION } from '../../../next-server/lib/constants'
+
+const PLUGIN_NAME = 'TraceEntryPointsPlugin'
+const TRACE_IGNORES = [
+  '**/*/node_modules/react/**/*.development.js',
+  '**/*/node_modules/react-dom/**/*.development.js',
+]
+
+function getChunkGroupFromBlock(
+  compilation: any,
+  block: any
+): webpack.compilation.ChunkGroup {
+  if (isWebpack5) {
+    return compilation.chunkGraph.getBlockChunkGroup(block)
+  }
+
+  return block.chunkGroup
+}
+
+function getModuleFromDependency(
+  compilation: any,
+  dep: any
+): webpack.Module & { resource?: string } {
+  if (isWebpack5) {
+    return compilation.moduleGraph.getModule(dep)
+  }
+
+  return dep.module
+}
+
+export class TraceEntryPointsPlugin implements webpack.Plugin {
+  private appDir: string
+  private entryTraces: Map<string, string[]>
+  private entryModMap: Map<string, any>
+  private entryNameMap: Map<string, string>
+
+  constructor({ appDir }: { appDir: string }) {
+    this.appDir = appDir
+    this.entryTraces = new Map()
+    this.entryModMap = new Map()
+    this.entryNameMap = new Map()
+  }
+
+  createTraceAssets(compilation: any, assets: any) {
+    const namePathMap = new Map<string, string>()
+
+    this.entryNameMap.forEach((value, key) => {
+      namePathMap.set(value, key)
+    })
+
+    for (const entrypoint of compilation.entrypoints.values()) {
+      const entryFiles = new Set<string>()
+      const entryMod = this.entryModMap.get(namePathMap.get(entrypoint.name)!)
+
+      entrypoint.getFiles().forEach((file: string) => {
+        if (!file.endsWith(entrypoint.name + '.js')) {
+          entryFiles.add(nodePath.join(compilation.outputOptions.path, file))
+        }
+      })
+
+      for (const block of entryMod.blocks) {
```

Comment:
This will only find `import()` in the entry module, does it?

Instead of following modules, you can ignore the whole module graph and follow the chunk graph instead.

There is a helper to get all referenced chunks: `Chunk.getAllReferencedChunks`

`entrypoint.getEntrypointChunk().getAllReferencedChunks()` should give you the expected result.

``` js
const allFiles = new Set();
for(const chunk of entrypoint.getEntrypointChunk().getAllReferencedChunks()) {
  for(const file of chunk.files) allFiles.add(file);
  for(const file of chunk.auxiliaryFiles) allFiles.add(file);
  // ... (see other comment)
}
```